gremlin/templates/chao-deployment.yaml

```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/instance: chao
     app.kubernetes.io/name: chao
     app.kubernetes.io/version: "1"
+    {{- toYaml .Values.extraMetadataLabels | trimSuffix "\n" | nindent 4 }}
```

Comment:
It sounds like you want something like `commonLabels` that are assigned to all components, which include the Helm and `app.kubernetes.io/*` labels.

For example, if you run `helm create example`, this will generate a Helm chart called `example`. In `templates/_helpers.tpl` you would find the following:

```

{{/*
Common labels
*/}}
{{- define "example.labels" -}}
helm.sh/chart: {{ include "example.chart" . }}
{{ include "example.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end }}

{{/*
Selector labels
*/}}
{{- define "example.selectorLabels" -}}
app.kubernetes.io/name: {{ include "example.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
```

You can then use them in helm charts as `example.labels`, which would include all the relevant labels. For example, for usage in a deployment:
```
# deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "example.fullname" . }}
  labels:
    {{- include "example.labels" . | nindent 4 }}
spec:
  {{- if not .Values.autoscaling.enabled }}
  replicas: {{ .Values.replicaCount }}
  {{- end }}
  selector:
    matchLabels:
      {{- include "example.selectorLabels" . | nindent 6 }}
...
```

Reply:
Thanks, that looks much nicer! 

In order to keep this PR well scoped, I didn't modify existing labels (e.g. `app.kubernetes.io`). However, I refactored the labels I added to use a common template. In the future, other labels can be migrated. I kept the "extra" map, since some of the keys / values can be specific to a certain company / application. However, I like this structure more, since it is now easier to add / remove labels from all resources.